### DISCUSSION TITLE
waveform overflows container

### DISCUSSION THREAD

**Discussion by shulinnn** [katspaugh/wavesurfer.js#3221] [Q&A] [answered]
Hello. I'm trying to figure out how to fit my waveform inside the container only so its doesnt overflow.
Is it because bigger audio files(longer waveform) cant render so small ? How would i go about responsivity then ?
Thanks in advance

```
  <div class="flex justify-center items-center h-screen">
    <div class="max-w-xs grid grid-flow-row py-3 bg-white rounded-full overflow-hidden shadow-lg">
      <div id="player" class=""></div>
    </div>
  </div>
```

```
const wavesurfer = w.create({
  container: "#player",
  waveColor: "black",
  progressColor: "gray",
  minPxPerSec: 5,
  fillParent: true,
  barWidth: 2,
  url: a,
  height: 50,
  autoCenter: true,
  autoScroll: true,
});

```

![image](https://github.com/katspaugh/wavesurfer.js/assets/49112352/092fd5a2-8a90-41d5-a96b-11bc2105449a)

**Comment by katspaugh** [comment, marked as answer]
Wavesurfer doesn't have its own width, it always fills up the container. So if you place your container in a flex/grid box, you need to give it a width.

**Reply by shulinnn** [reply]
I thought thats what i'm doing.

```
    <div class="wrapper">
      <div id="player" class=""></div>
    </div>
```

```
.player-container > .wrapper {
  @apply max-w-xs;
  @apply w-80;
  @apply py-3;
  @apply grid;
  @apply grid-flow-row;
  @apply bg-white;
  @apply rounded-full;
  @apply overflow-hidden;
  @apply shadow-lg;
}
```

**Reply by shulinnn** [reply]
Oh yes , i forgot what my actual container for the waveform is , thanks a lot , lol.
